packages/cadl-ranch-expect/src/decorators.ts

```diff
@@ -133,8 +133,20 @@ function getRouteSegments(program: Program, target: Operation | Interface | Name
 }
 
 function getOperationRoute(program: Program, target: Operation): string {
+  const template = getHostTemplate(program, target.namespace);
   const segments = getRouteSegments(program, target);
-  return "/" + segments.map((x) => (x.startsWith("/") ? x.substring(1) : x)).join("/");
+  return (template ? template : "/") + segments.map((x) => (x.startsWith("/") ? x.substring(1) : x)).join("/");
+}
+
+function getHostTemplate(program: Program, namespace?: Namespace): string | undefined {
+  if (namespace === undefined) {
+    return undefined;
+  }
+  const server = getServers(program, namespace);
+  if (server && server.length === 1) {
+    return server[0].url.split("localhost:3000")[1];
```

Comment:
```suggestion
    return new URL(server[0].url).path
```
Or whatever else is the correct property on the URL object for the entire path
I think that would be more reliable.

Reply:
Actually we can't, there's a case like `{endpoint}//server/path/multiple/{apiVersion}` and if we use URL, it will report invalid url errors.

Reply:
hhm right, could maybe catch the error and then keep the entire server url